### PR TITLE
ZTP SiteConfig & PreReq added

### PR DESCRIPTION
**Pull request by hhamid93**
ZTP base has been created with the following yaml added:

[base/ztp/bmc-secrets.yaml]
[base/ztp/kustomization.yaml]
[base/ztp/pullsecret-sealed.yaml]
[overlay/ztp/SiteConfig-example-3node.yaml]

### CODE REVIEW

**Review by natifridman — COMMENTED** [review]
Since ZTP contains also PolicyGenTemplate, maybe this should be part of a new folder inside `ztp`? something like `base/ztp/site-config`?

Also, please fix the yaml-lint warnings

File: overlay/ztp/site-config/kustomization.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
+---
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+
+generators:
+# - SiteConfig-example-sno.yaml
+# - SiteConfig-example-3node.yaml
+# - SiteConfig-example-standard.yaml
```
Comment (natifridman) [inline overlay/ztp/site-config/kustomization.yaml]: Why is this in a comment?

File: overlay/ztp/site-config/SiteConfig-example-sno.yaml
Comment (natifridman) [inline overlay/ztp/site-config/SiteConfig-example-sno.yaml:1]: The Xs here because this is just an example? I don't see patches for this

Comment (hhamid93) [reply overlay/ztp/site-config/SiteConfig-example-sno.yaml:1]: yes! So for SiteConfig CR, There are no patches for SiteConfig, The SiteConfig yaml goes under generators as it is. Maybe writing a README to elaborate on this won't be a bad Idea. Will get it going.

Comment (hhamid93) [reply overlay/ztp/site-config/SiteConfig-example-sno.yaml:1]: added README

**Review by natifridman — APPROVED** [review]
LGTM

**Pull request merged**